core/src/main/scala/org/broadinstitute/dsde/rawls/entities/EntityService.scala

```diff
@@ -664,10 +665,15 @@ class EntityService(protected val ctx: RawlsRequestContext,
 
         // finally, change the workspace to be quicksilver-enabled
         _ <- traceFutureWithParent("setWorkspaceSettings", s) { _ =>
-          workspaceSettingService.setWorkspaceSettings(
-            workspaceName,
-            List(CompactDataTablesSetting(CompactDataTablesConfig(enabled = true)))
-          )
+          if (!updateSettings) {
+            // if this is a settings migration, we don't want to set the setting again
+            Future.successful(())
+          } else
+            // otherwise, set the CompactDataTablesSetting to enabled
+            workspaceSettingService.setWorkspaceSettings(
+              workspaceName,
+              List(CompactDataTablesSetting(CompactDataTablesConfig(enabled = true)))
+            )
```

Comment:
you'll also need to handle the logic early in this method; search for "confirm if this is already a quicksilver workspace by checking settings".

This makes me somewhat uncomfortable - I haven't tried it, but I think that if you 1) migrate a workspace from legacy to Quicksilver with cleanup enabled, then 2) re-run the migration on the same workspace, it will end up erasing all your attributes. If you do that without cleanup enabled, it will revert your attributes to the last data saved to legacy, erasing any Quicksilver changes you've made since then.